### DISCUSSION TITLE
reading in every Nth event from a file

### DISCUSSION THREAD

**Discussion by calebmiller** [scikit-hep/uproot5#287] [Q&A] [answered]
I'd like to load a TTree from a rootfile but with a prescale applied. ie. Instead of loading each event into the array I only load events where event_id(mod)10=0 or event_id(mod)100=0. 

I imagine this is fairly easy to do if the event_id is already in the TTree as a branch, by just requiring the event meets the mod 0 requirement. But what about if there is no explicit event number branch?

**Comment by jpivarski** [comment, marked as answer]
Any array in Python can use a slice syntax like this:

```python
array[::10]
```

The above will take event `0`, event `10`, event `20`, etc. (If you wanted event `9`, `19`, `29`, etc., then you can do `array[9::10]`. That would agree better with `eventId % 10` if the `eventId` started with `1` instead of `0`.)

There's no way to avoid _reading_ all the events before applying this prescale because contiguous events have to be read together (due to the columnar structure of a ROOT TBranch), but this will get you every 10th event after they've been read.

**Reply by calebmiller** [reply]
Ah that makes sense. Thanks for your help
